ls1intum/Artemis
# [Exam] Bugfix: exam visibility and disabled input in start page

### Checklist

### Motivation and Context


The logic for the check if there is a visible exam was faulty. See: https://github.com/ls1intum/Artemis/issues/1734

For the start page you could still change your name in the input field after you clicked on 'Start Exam'
### Description

Adjusted the hasVisibleExams to check if there are really visible exams
Added the server time sync for date checks in the exam overview 

Disabled input field when waiting for the exam to start

### Steps for Testing


Try to reproduce the behaviour in https://github.com/ls1intum/Artemis/issues/1734 and confirm that it is working as intended

Go to an exam and see if you can still change your name after clicking on 'Start Exam' button
